core/broker/broker.go

```diff
@@ -18,25 +18,12 @@ import (
 	"sync"
 	"time"
 
+	"code.vegaprotocol.io/vega/shared/broker"
+
 	"code.vegaprotocol.io/vega/core/events"
 	"code.vegaprotocol.io/vega/logging"
 )
 
-// Subscriber interface allows pushing values to subscribers, can be set to
-// a Skip state (temporarily not receiving any events), or closed. Otherwise events are pushed
-//
-//go:generate go run github.com/golang/mock/mockgen -destination mocks/subscriber_mock.go -package mocks code.vegaprotocol.io/vega/core/broker Subscriber
-type Subscriber interface {
```

Comment:
Interfaces should be declared in the package that uses/relies on them. By moving this to a shared package, the broker in datanode and core become tightly coupled in a way that they shouldn't be. Let's say that we want to change snapshots in core to work asynchronously, using subscribers, and want to change the interface, that should not impact the interface on the data node side of things.
The broker interface so far was the only exception, and we only made that change after having discussed the matter rather extensively.

IMO, because the broker in data node and core are separate entities, and the role of their subscribers may diverge in the future, I don't think this shared package is a good idea.

Reply:
can you take that up with @jeremyletang, questioned was asked and  agreed they should be consolidated (see original issue)

Reply:
as discussed the reason given against this change, i.e. -> 'We should follow "the go proverbs", specifically: a little bit of copying is better than a a little dependency.' is fine for trivial bits of code, but this is clearly not that, and is actually a perfect example of why we should not duplicate code willy nilly, it was pure chance that I stumbled across the duplicate code, had I not we would have been in a situation where the mem leak would have been fixed in datanode but not in core.